packages/versioning/src/versioning.ts

```diff
@@ -58,13 +58,24 @@ export function $removed(context: DecoratorContext, t: Type, v: EnumMember) {
   }
   program.stateMap(removedOnKey).set(t, version);
 }
+
+interface RenamedFrom {
+  v: Version;
+  oldName: string;
+}
+
 export function $renamedFrom(context: DecoratorContext, t: Type, v: EnumMember, oldName: string) {
   const { program } = context;
   const version = checkIsVersion(context.program, v, context.getArgumentTarget(0)!);
   if (!version) {
     return;
   }
-  const record = { v: version, oldName: oldName };
+
+  // retrieve statemap to update or create a new one
+  const record = program.stateMap(renamedFromKey).get(t) ?? new Array<RenamedFrom>();
+  record.push({ v: version, oldName: oldName });
+  // ensure that records are stored in ascending order
+  (record as Array<RenamedFrom>).sort((a, b) => a.v.index - b.v.index);
```

Comment:
```suggestion
  record.sort((a, b) => a.v.index - b.v.index);
```

this sorts in place so it can have unexpected consequences. Would copy the array first with `[...record]`

Reply:
An in-place sort is what I intended here. Is there a reason I'd want to make a copy first?

Reply:
well here you are getting the data from the program state and then sorting than in place so updating whats in the program state and so affecting other user

Reply:
Yes, but I explicitly want to ensure that what's in program state here is sorted.

Reply:
should we sort on insert then?

Reply:
nvm that's what is happening here...

Reply:
all good then